src/services/handler/__test__/broker.test.js

```diff
@@ -1,62 +1,74 @@
-import { INTERACTION_IDS } from '../../../constants/interaction-ids';
-import continueMessageHandler from "../continue-message-handler";
+import { readFileSync } from "fs";
+import expect from "expect";
+import path from "path";
 
+// Mocking
 jest.mock('../ehr-request-handler');
 jest.mock('../continue-message-handler');
+jest.mock('../acknowledgement-handler');
+jest.mock('../../parser/parsing-utilities');
 jest.mock('../../../middleware/logging');
 
 describe('broker', () => {
-  it('should hand off to ehr-request-handler if it is an EHR request', async () => {
-    let ehrRequest = {
-      interactionId: INTERACTION_IDS.EHR_REQUEST_INTERACTION_ID,
-      conversationId: '17a757f2-f4d2-444e-a246-9cb77bef7f22',
-      ehrRequestId: 'FFFB3C70-0BCC-4D9E-A441-7E9C41A897AA',
-      nhsNumber: '9692842304',
-      odsCode: 'A91720'
-    };
+  it('should hand off to the ehr-request-handler if it is a EHR request', async () => {
+    // given
+    const ehrRequest = readFileSync(path.join(__dirname, "data", "ehr-requests", "RCMR_IN010000UK05"), "utf-8");
+
+    // when
+    parseInteractionId.mockResolvedValueOnce(Promise.resolve(INTERACTION_IDS.EHR_REQUEST_INTERACTION_ID));
 
     await sendMessageToCorrespondingHandler(ehrRequest);
 
+    // then
     await expect(ehrRequestHandler).toHaveBeenCalledWith(ehrRequest);
   });
 
-  it('should hand off to continue-message-handler if it is a continue request', async () => {
-    let continueRequest = {
-      interactionId: INTERACTION_IDS.CONTINUE_REQUEST_INTERACTION_ID,
-      conversationId: '27a757f2-f4d2-444e-a246-9cb77bef7f22',
-      nhsNumber: '8692842304',
-      odsCode: 'B91720'
-    };
+  it('should hand off to the continue-message-handler if it is a continue request', async () => {
+    // given
+    const continueRequest = readFileSync(path.join(__dirname, "data", "continue-requests", "COPC_IN000001UK01"), "utf-8");
+
+    // when
+    parseInteractionId.mockResolvedValueOnce(Promise.resolve(INTERACTION_IDS.CONTINUE_REQUEST_INTERACTION_ID));
 
     await sendMessageToCorrespondingHandler(continueRequest);
 
+    // then
     await expect(continueMessageHandler).toHaveBeenCalledWith(continueRequest);
   });
 
-  it('should log if it is an acknowledgement response', async () => {
-    let acknowledgementResponse = {
-      interactionId: INTERACTION_IDS.ACKNOWLEDGEMENT_INTERACTION_ID,
-      conversationId: '37a757f2-f4d2-444e-a246-9cb77bef7f22',
-      nhsNumber: '7692842304',
-      odsCode: 'C91720'
-    };
+  it('should hand off to the acknowledgement-handler if it is a negative acknowledgement', async () => {
+    // given
+    const negativeAcknowledgement = readFileSync(path.join(__dirname, "data", "acknowledgements", "negative", "MCCI_IN010000UK13_TPP_AR_01"), "utf-8");
 
-    await sendMessageToCorrespondingHandler(acknowledgementResponse);
+    // when
+    parseInteractionId.mockResolvedValueOnce(Promise.resolve(INTERACTION_IDS.ACKNOWLEDGEMENT_INTERACTION_ID));
```

Comment:
`INTERACTION_IDS.ACKNOWLEDGEMENT_INTERACTION_ID` probably should be shortened to `INTERACTION_IDS.ACKNOWLEDGEMENT` since we are already aware it's an Interaction ID.